### ISSUE TITLE
Device or Resource Busy

### ISSUE THREAD

**Issue by elliwigy** [astarasikov/exynos9610-edl-mode#2]
Hello,

I have been trying for a long time to fix my bricked A505U. I took your tool and compiled based on the instructions as well as took the stock sboot from stock rev1 firmware for my device and used your script to split the sboot into parts. I get the below:

SMDK42XX USB Download Tool
Version 0.20 (c) 2004,2005,2006Ben Dooks <ben-linux@fluff.org>

=> found device: bus 001, dev 019
=> loaded 8192 bytes from part1.bin
usb_set_configuration: Device or resource busy
SMDK42XX USB Download Tool
Version 0.20 (c) 2004,2005,2006Ben Dooks <ben-linux@fluff.org>

=> found device: bus 001, dev 019
=> loaded 77824 bytes from part2.bin
usb_set_configuration: Device or resource busy
SMDK42XX USB Download Tool
Version 0.20 (c) 2004,2005,2006Ben Dooks <ben-linux@fluff.org>

=> found device: bus 001, dev 019
=> loaded 192512 bytes from part3.bin
usb_set_configuration: Device or resource busy
SMDK42XX USB Download Tool
Version 0.20 (c) 2004,2005,2006Ben Dooks <ben-linux@fluff.org>

=> found device: bus 001, dev 019
=> loaded 1572864 bytes from part4.bin
usb_set_configuration: Device or resource busy
SMDK42XX USB Download Tool
Version 0.20 (c) 2004,2005,2006Ben Dooks <ben-linux@fluff.org>

=> found device: bus 001, dev 019
=> loaded 262144 bytes from part5.bin
usb_set_configuration: Device or resource busy
SMDK42XX USB Download Tool
Version 0.20 (c) 2004,2005,2006Ben Dooks <ben-linux@fluff.org>

=> found device: bus 001, dev 019
=> loaded 1052672 bytes from part6.bin
usb_set_configuration: Device or resource busy

As you can see, any parts I attempt to flash are loaded, it sees my device in Exynos9610 mode but doesn't seem to be able to properly flash the parts.

Keep in mind it does similar when I try to use Windows and the multidownloader tool with a cfg file and the parts as well. it's like the first part starts to flash then it just does nothing.

Any ideas?

To add, I got into Exynos USB mode by renaming the cm.bin to sboot and flashing in ODIN. I am an android developer/researcher and was trying to bypass factory token but ultimately the recovery files/parts didn't work so it has been stuck in this mode for a while.

Thanks in advance!

**Comment by astarasikov** [comment]
@elliwigy the image you're trying to download needs to have the same or higher rollback number as the one you've flashed previously. Also, try a different USB port (if you're running this on a PC, not on a macbook).

**Comment by elliwigy** [comment]
Yes, I am very familiar with rollback prevention and that its not possible to downgrade.

The device was purchased and bricked before anything beyond rev1 firmware was even available.

I have tried multiple sboots from multiple firmwares as well as multiple ports, hubs, etc.

I am starting to feel that the U model is unique. Do you feel its possible they have some sort of verification or signature checks for usa variants?

For example the SD counterparts in EDL mode require the firehose programmers to be signed/official. Not sure if there is any easy way to check this. Other than that I am at a loss of what to try next.

**Comment by astarasikov** [comment]
I don't think "usb_set_configuration" matters, it can be even commented out. If it's behaving the same way on Windows, then it's either a hardware issue or the offsets for splitting the s-boot.bin are wrong.
Please reach out to me privately if you want to discuss this further.

**Comment by Bimo754** [comment]
Hey @astarasikov , @elliwigy I am trying with the device Motorola One Vision, any progress?
